### PR TITLE
fix: remove har-validator

### LINKED ISSUE #9: har-validator is no longer supported
https://www.npmjs.com/package/har-validator

node version 16.13.0
npm version 8.1.0

### PR DESCRIPTION
**Pull request by JJ**
## PR Checklist:
- [x] I have run `npm test` locally and all tests are passing. *Also yarn install*, which complains about all the stuff in #12 mostly
- [x] I have added/updated tests for any new behavior. (Actually, added a GitHub actions to avoid from changes being merged if they fail with `yarn`)
- [x] If this is a significant change, an issue has already been created where the problem / solution was discussed: This supersedes #15 which by eliminating the only code that uses it, as well as the place where it's used

## PR Description

#11 was introduced to remove at least one of the outdated dependencies, with several vulnerabilities, that were being reported as far back as 2020 (cypress-io/cypress#8265) in several issues here and in `cypress`. Check out #17, for instance.
This PR eliminates the need to use `har-validator` by eliminating the code that used it, that was not tested, and in fact HTTP archive validation does not seem like the kind of thing that `cypress` will need to support. 

This 
* closes #9 
* supersedes #15 
*  solves cypress-io/cypress#19097 and cypress-io/cypress#19102
* solves #14 (and prevents from it happening in the future)

### CODE REVIEW

**Review by jamie-phlo — APPROVED** [review]

**Review by ceIia — APPROVED** [review]

**Review by erwanriou — APPROVED** [review]
ok this also work

**Review by JanKanty — APPROVED** [review]

**Review by niltonxp2 — APPROVED** [review]

**Review by dceballos24 — APPROVED** [review]

**Review by derBinder — APPROVED** [review]

**Review by damikun — APPROVED** [review]

**Review by AntoineKoeman — APPROVED** [review]

**Review by MichaelPhan — APPROVED** [review]

**Review by sbiliaiev — APPROVED** [review]

**Comment by meeroslav** [comment]
This makes more sense than #15

**Review by larionov — APPROVED** [review]

**Review by rmoralp — APPROVED** [review]

File: .github/workflows/yarn.yaml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+name: Check with yarn
```
Comment (meeroslav) [inline .github/workflows/yarn.yaml:1]: I would move this to a separate PR and extend it to test multiple package managers via GitHub Action's matrix, if that's the purpose of this workflow.

That would allow the PR to contain only atomic change and be more likely merged in.

**Review by meeroslav — CHANGES_REQUESTED** [review]

File: .github/workflows/yarn.yaml (hunk 1 above)
Comment (JJ) [reply .github/workflows/yarn.yaml:1]: Can certainly do that if required; but in this case I felt this was necessary, since it seems to be the way to check if it's working with the alternate package manager. The existing Circle CI only runs the tests with npm, and didn't catch the error before.

Comment (meeroslav) [reply .github/workflows/yarn.yaml:1]: Interesting. Our CircleCI runs failed on npm, yarn and pnpm equally.

Comment (JJ) [reply .github/workflows/yarn.yaml:1]: You mean with #13 or after #13 was merged. It apparently failed also if package-lock.json was not present?
Anyway, I think there's no real reason why there were no GitHub actions here other than this module needing some love from the community. After all, it didn't even have a package-lock.json until #13

**Review by meeroslav — APPROVED** [review]

**Comment by JJ** [comment]
Ping?

**Review by flotwig — CHANGES_REQUESTED** [review]

File: .github/workflows/yarn.yaml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+    name: Install and test with yarn
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: install
+      - uses: borales/actions-yarn@v2.3.0
+        with:
+          cmd: test
```
[the remark this answers is not in the document]
Comment (flotwig) [reply .github/workflows/yarn.yaml:14]: Seems to work: https://github.com/cypress-io/request/runs/4356870266?check_suite_focus=true

**Review by flotwig — APPROVED** [review]
@JJ thanks for fixing this up.
